core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -934,6 +942,33 @@ public String[] getUserListOfJDBCRole(RoleContext ctx, String filter, int maxIte
         return names;
     }
 
+    /**
+     * Return the count of users belong to the given role for the given {@link RoleContext}.
+     *
+     * @param ctx {@link RoleContext} corresponding to the role.
+     * @throws UserStoreException If an unexpected error occurs in user store.
+     */
+    public int getUserCountByRole(RoleContext ctx) throws UserStoreException {
+
+        String roleName = ctx.getRoleName();
+        Connection dbConnection = null;
+        String sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_USERS_COUNT_WITH_FILTER_ROLE);
+        try {
```

Comment:
Shall we try: try with resource?

Reply:
To be consistent with other db calls in user core, try method will be used with manual db close.